### ISSUE TITLE
zig build --help : "error: AccessDenied"

### ISSUE THREAD

**Issue by andy-preston** [ziglang/zig#19400]
### Zig Version

0.12.0-dev.3412+54c08579e

### Steps to Reproduce and Observed Behavior

FROM alpine:3.19.1 as zig-builder

Dockerfile:

```
FROM alpine:3.19.1 as zig-builder
ADD https://ziglang.org/builds/zig-linux-x86_64-0.12.0-dev.3412+54c08579e.tar.xz \
    /usr/local/share/zig.tar.xz
RUN cd /usr/local/share && \
    tar -xvf zig.tar.xz && \
    mv zig-linux-x86_64-0.12.0-dev.3412+54c08579e zig
ENV PATH="/usr/local/share/zig:$PATH"
```

shell commands:
```
docker build --tag zig docker
docker run --rm --interactive --tty \
    --volume ./test:/usr/local/src \
    --workdir /usr/local/src \
    --user $(id -u):$(id -g)  \
    zig sh
```

Inside docker:
```
zig init
zig build --help
```

Response:
`error: AccessDenied`

### Expected Behavior

My guess from experience is that this wants to set up a file in my home directory... running in Docker, I obviously have no home directory... but actually saying (made up examples as I'm left guessing) "Access to ~/.zig/settings/something denied" would be more helpful than the current "dead-end" error message. Some clue as to what has gone wrong would help.

**Comment by mikdusan** [comment]
Agreed. BTW, it's because global cache is expected to be in home by default.

- workaround: `export ZIG_GLOBAL_CACHE_DIR=$PWD`

**Comment by kassane** [comment]
To solve the access problem in Alpine, I needed to create a user.
https://github.com/kassane/docker-zig/blob/9b24ba6c979c3aa8fa6835d5eeaf1613ca219f1f/alpine/master/Dockerfile#L45

**Comment by boymaas** [comment]
Also ran into this error when trying to do work packaging with nixos, took me a while to figure out.
